base/Dockerfile

```diff
@@ -0,0 +1,38 @@
+FROM phusion/passenger-ruby26:1.0.10
+
+RUN echo 'Downloading Packages' && \
+  curl -sL https://deb.nodesource.com/setup_12.x | bash - && \
+  curl -sL https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - && \
+  echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list && \
+  apt-get update -qq && \
+  apt-get install -y --no-install-recommends \
+    build-essential \
+    libsasl2-dev \
+    nodejs \
+    rsync \
+    tzdata \
+    yarn \
+  && \
+  apt-get clean && \
+  rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* && \
+  echo 'Packages Downloaded'
+
+# Install Chrome so we can run system specs for Blacklight
+RUN apt-get update && apt-get install -y wget && \
+  wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - && \
+  sh -c 'echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google-chrome.list' && \
+  apt-get update && \
+  apt-get install -y google-chrome-stable && \
+  apt-get clean && \
+  rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* && \
+  echo 'Chrome installed'
+
+RUN yarn && \
```

Comment:
I think the main thing that concerns me about this refactor is that it *seems* like it assumes that the Blacklight and Management applications will have non-conflicting dependencies. If people with more experience don't see that as a blocker, that's cool, but it concerns me that it will create non-obvious snarls with dependencies on the applications. One thing that would help me feel more confident about that is if the Blacklight and Management builds on top of this base were tested in this repo.

Reply:
This assumes a common base and strongly encourages both projects to keep that commonality.  That doesn't mean that one project can't add an additional dependency if they need to, but that should be rare and should be discouraged, especially if it is the case that the two projects need conflicting dependencies. That sort of environmental inconstancy breeds bugs and lack of interoperable code. It creates another layer of domain knowledge that developers need to switch between projects and it is, in my experience, rarely worth the gain. 

Since the upstream images can not exist with out a base to point to, their tests essentially _are_ a test of this base and I don't really understand building an additional suite. One thing I am concerned about is this location for this code. Right now Camerata brings the two projects together for deployment. This provides a common starting place for both of them. There's some niceness to that being together, but it also could be seen as creating a dependency loop (camerata depends on a build of management and management depends on a build of base (which is part of camerata)). I'm not sure that prevents this from being merged here and then moved later, but it is something I want to point out as worth considering.

Reply:
Ah, I think I was misremembering that in the "base" step we were copying and installing all the gem and yarn dependencies themselves, which was what I was worried about. If I read *this* code, it's not doing that, so it's not a problem.